### PR TITLE
new datasets + query reformulators

### PR DESCRIPTION
**Pull request by AlexandreMisrahi2005**
- new datasets: bioasq-pubmed, coderagbench (humaneval + mbpp), apibench, syllabusqa)
- add query generators/reformulators to reformulate query with generator @nadiinchi 
- update generators(wrapper to load gguf format + wrapper to load local model)
- some assertions (check label format == list, and id format == string)
- generate oracles scripts (bioasq + humaneval)

Please check especially the files ```utils.py``` and ```modules/metrics.py``` where there could be some potentially critical changes

### CODE REVIEW

**Comment by nadiinchi** [comment]
My review

* create a folder in `config/datasets` with all new datasets? e.g. `multi-domain` ?
* same suggestion for `config/prompts`
* put all new domain dataset processors into a separate file in `processors`, as it is done for KILT? (i'd suggest both datastores and queries)
* code rag bench: add separate configs per doc dataset, cause they need to be run separately before using MergedDocDataset
* remove config `config/generator/llama-3-8b-instruct-FT-NQ-2epochs.yaml` 
* `config/query_generator/unfold_api_query.yaml`: are all commented parts needed?
* train code: is it possible to add optional support of wandb? for now its imported mandatory. or discuss this with other contributors
* train code: could we make num_saving_steps  a config argument and not hardcoded?

**Comment by AlexandreMisrahi2005** [comment]
All comments from @nadiinchi are taken into account in the new commits

**Comment by nadiinchi** [comment]
* move  `init_generator` function into `models/query_generator.py` ("abstract" class for query generation)
* remove unnecessary imports (urllib and bs4) from `modules/dataset_processors.py`
* move `bs4` import into the corresponding class(es) in `modules/processors/multidomain_dataset_processor.py`

Everything else looks good to me

File: models/generators/vllm.py
Comment (sclincha) [inline models/generators/vllm.py:1]: remove comment

File: scripts/humaneval_generate_oracle.py
Comment (sclincha) [inline scripts/humaneval_generate_oracle.py:1]: delete

File: scripts/bioasq_generate_oracle.py
Comment (sclincha) [inline scripts/bioasq_generate_oracle.py:1]: to keep and update by @DRRV

**Pull request merged**